### PR TITLE
Radar - Added setting to change priority for icons

### PR DESCRIPTION
**Pull request by johnb432**
Selection now is either "Engineer, Medic & EOD" or "AR & AT".
Addresses #189.
I haven't tested it however.

### CODE REVIEW

File: addons/radar/XEH_preInit.sqf [hunk 1]
```diff
@@ -466,6 +466,18 @@ if (isClass(configfile >> "CfgPatches" >> "diwako_dui_buddy")) then {
     diwako_dui_namelist_only_buddy_icon = false;
 };
 
+[
+    QGVAR(icon_priority_setting)
+    ,"LIST"
+    ,[localize "STR_dui_namelist_icon_priority", localize "STR_dui_namelist_icon_priority_desc"]
+    ,[CBA_SETTINGS_CAT, _curCat]
+    ,[[0, 1], [localize "STR_dui_namelist_icon_priority_setting0", localize "STR_dui_namelist_icon_priority_setting1"], 1]
+    ,false
```
Comment (johnb432) [inline addons/radar/XEH_preInit.sqf:475]: Unsure if this setting has to be enforced globally or not.

File: addons/radar/XEH_preInit.sqf [hunk 2]
```diff
@@ -466,6 +466,18 @@ if (isClass(configfile >> "CfgPatches" >> "diwako_dui_buddy")) then {
...
+    ,{
+        [QGVAR(refreshUI),[]] call CBA_fnc_localEvent;
```
Comment (johnb432) [inline addons/radar/XEH_preInit.sqf:477]: Also unsure if this is needed.

Comment (diwako) [reply addons/radar/XEH_preInit.sqf:477]: not needed

File: addons/radar/XEH_preInit.sqf (hunk 1 above)
Comment (diwako) [reply addons/radar/XEH_preInit.sqf:475]: While I think the icon style should be a client side setting, what the icon represents and on which position in the list it should be is not. I would say this should be a globally enforced setting.

**Comment by diwako** [comment]
Been thinking the getIcon function needs a refactor, it needs to accommodate several things now. Just not sure how at the moment.

Right now tho, you should probably test this first if it works. The refactor is something I need to think about once i got more time on my hands.

**Review by diwako — CHANGES_REQUESTED** [review]
As discussed above. 2 small changes should be made.

File: addons/radar/XEH_preInit.sqf (hunk 2 above)
Comment (johnb432) [reply addons/radar/XEH_preInit.sqf]: Removed.

File: addons/radar/XEH_preInit.sqf (hunk 1 above)
Comment (johnb432) [reply addons/radar/XEH_preInit.sqf]: Changed.

**Comment by johnb432** [comment]
Still need to test, will hopefully get around to doing it sometime soon.

**Comment by johnb432** [comment]
I have been able to test it, although to a limited but maybe sufficient scale:
One squad with AI in the ACE Arsenal and an MP mission with 2 players total.
In both cases it worked as intended, no side effects I could notice. Setting can be change mid mission without breaking stuff, as far as I can tell.
Leaving and rejoining a squad seemed to not break anything.

If you have any edge case scenarios you know of, I _could_ try to test them, but I can't guarantee that I have the resources to do so.
Otherwise, it's good to go on my end.

**Comment by diwako** [comment]
Edge cases with the getIcon functions are not people joining or leaving the squad. The edge cases are if a unit is in a vehicle and the roles a unit can be in.

I will myself take a look at this change later and see if there is something to reconsider.

File: addons/main/stringtable.xml [hunk 3]
```diff
@@ -382,6 +382,18 @@
             <Spanish>Muestra solo el icono de amigo en la lista de nombres en lugar de ambos, amigo y clase de unidad</Spanish>
             <Russian>Показывает только иконку в списке вместо имени и класса юнита</Russian>
         </Key>
+        <Key ID="STR_dui_namelist_icon_priority">
+            <English>Prioritise icons:</English>
```
Comment (diwako) [inline addons/main/stringtable.xml:386]:
```suggestion
            <English>Prioritise icons</English>
```
the `:` is already added by the ba settings framework

**Review by diwako — CHANGES_REQUESTED** [review]
Just tested it, works as expected. Found only one very minor thing. Also thank you for setting the setting to reflect the current icon preferences

**Review by jokoho48 — APPROVED** [review]
